File: 정선웅/주식가격.java
import java.util.*;

class Solution {

    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];

        for(int i=0; i<prices.length-1; i++) {
            int num = 0;
            for(int j=i+1; j<prices.length; j++) {
                num++;
                if(prices[i] > prices[j]) {
                    break;
                }
            }
            answer[i] = num;
        }
        return answer;
    }
}

File: 표/stack_queue_heap/주식가격.py
from collections import deque

def solution(prices):
    answer = []
    queue = deque(prices)
    while queue:
        price = queue.popleft()
        time = 0
        for p in queue:
            if p < price:
                time += 1
                break
            time += 1
        answer.append(time)
    
    return answer